### DISCUSSION TITLE
Slow Touchscreen Response question

### DISCUSSION THREAD

**Discussion by Bwanna** [ImpulseAdventure/GUIslice#483] [Q&A]
Hi,
What a great project GUISLice is making display work so much easier.  Thanks for this! I just have a question about how to increase touchscreen response time with an STMPE610.  

My working project reads 7 sensors via two ADC1115 devices and processes the readings for display via GUISlice.  The display includes RadioButtons that changes various display presentations, such as, temperature between F and C and 12h/24h clock setting.

The sketch is rather large and cycles constantly through the measuring/processing of the sensors, eating up plenty of MCU time... and likely the reason for delayed response to touchscreen actions.  I'm looking at changing this to maybe reading/processing sensors once per second instead of continuously.

My question:  Is there a method of using the STMPE610 interrupt to affect a quick response?  

Thanks,
Steve

**Comment by Pconti31** [comment]
@Bwanna Not that I'm aware of. 

Your idea of checking sensors less often seems like the best approach but 7 sensors reading each time through your loop() doesn't seem like much overhead.  Not enough to slow performance of your touch screen.

Although the ADS1115 is not very fast - from the datasheet 'The ADS111x can perform conversions at data rates up to 860 samples per second (SPS).' i.e. 1.16 ms.  Which isn't awful.  So, Now the real issue is what library are you using to read the ADC1115?

If Adafruit's Adafruit_ADS1X15 library you need to use  'getDataRate' to determine default sample rate and 'setDataRate' commands to boost sample rate speed.
```
#define RATE_ADS1115_8SPS (0x0000)   ///< 8 samples per second
#define RATE_ADS1115_16SPS (0x0020)  ///< 16 samples per second
#define RATE_ADS1115_32SPS (0x0040)  ///< 32 samples per second
#define RATE_ADS1115_64SPS (0x0060)  ///< 64 samples per second
#define RATE_ADS1115_128SPS (0x0080) ///< 128 samples per second (default)
#define RATE_ADS1115_250SPS (0x00A0) ///< 250 samples per second
#define RATE_ADS1115_475SPS (0x00C0) ///< 475 samples per second
#define RATE_ADS1115_860SPS (0x00E0) ///< 860 samples per second
```
Maybe this could help. Also, not using floating point math will also speed things up.
Of course, if you are using an Arduino UNO not  much you can do...

Paul--

**Reply by Bwanna** [reply]
Appreciate the feedback.

My preference is to use the touchscreen INT to activate the screen response coding. Until I work that out, I have found a better touchscreen response by adjusting the rest of the coding.

My code has two main sections to it, one is sensor readings, two is information processing.  I've found a balance by having the information processing only done once every second, while the sensor readings occur continuously in the normal main loop. This is working well in that the sensor readings are up to date and available when the information processing occurs.
